src/components/content/deployedServices/myServices/MyServiceHistory.tsx

```diff
@@ -24,20 +27,80 @@ export const MyServiceHistory = ({
+    let taskStatusFilters: ColumnFilterItem[] = [];
+    let serviceOrdersHistoryList: ServiceOrderDetails[] = [];
+    const listServiceOrdersHistoryQuery = useListServiceOrdersHistoryQuery(deployedService.serviceId);
+
+    if (listServiceOrdersHistoryQuery.isSuccess && listServiceOrdersHistoryQuery.data.length > 0) {
+        serviceOrdersHistoryList = listServiceOrdersHistoryQuery.data;
+        updateOrderIdFilters(listServiceOrdersHistoryQuery.data);
+        updateTaskTypeFilters();
+        updateTaskStatusFilters();
+    }
 
-    if (listServiceModifyHistoryQuery.isSuccess && listServiceModifyHistoryQuery.data.length > 0) {
-        serviceModificationAuditHistoryList = listServiceModifyHistoryQuery.data;
+    function updateOrderIdFilters(resp: ServiceOrderDetails[]): void {
+        const filters: ColumnFilterItem[] = [];
+        resp.forEach((respItem) => {
+            const filter = {
+                text: respItem.orderId,
+                value: respItem.orderId,
+            };
+            filters.push(filter);
+        });
+        orderIdFilters = filters;
+    }
+
+    function updateTaskTypeFilters(): void {
+        const filters: ColumnFilterItem[] = [];
+        Object.values(taskType).forEach((taskTypeItem) => {
+            const filter = {
+                text: taskTypeItem,
+                value: taskTypeItem,
+            };
+            filters.push(filter);
+        });
+        taskTypeFilters = filters;
+    }
+
+    function updateTaskStatusFilters(): void {
+        const filters: ColumnFilterItem[] = [];
+        Object.values(taskStatus).forEach((taskStatusItem) => {
+            const filter = {
+                text: taskStatusItem,
+                value: taskStatusItem,
+            };
+            filters.push(filter);
+        });
+        taskStatusFilters = filters;
     }
 
     const columns: ColumnsType<ServiceOrderDetails> = [
         {
-            title: 'ModifyId',
+            title: 'OrderId',
             dataIndex: 'orderId',
             align: 'center',
+            width: 120,
+            className: serviceModifyStyles.modifyHistoryValue,
+            filters: orderIdFilters,
+            filterMode: 'tree',
+            filterSearch: true,
+            onFilter: (value: React.Key | boolean, record) => record.orderId.startsWith(value.toString()),
+        },
+        {
+            title: 'TaskType',
+            dataIndex: 'taskType',
+            align: 'center',
             width: 100,
             className: serviceModifyStyles.modifyHistoryValue,
+            filters: taskTypeFilters,
+            filterMode: 'tree',
+            filterSearch: true,
+            onFilter: (value: React.Key | boolean, record) => record.taskType.startsWith(value.toString()),
+            render: (value: taskType) => {
+                return <div className={serviceModifyStyles.serviceHistoryValue}>{value}</div>;
+            },
         },
         {
             title: 'Previous',
```

Comment:
I would suggest, that we remove the previous and new column and replace it with details column.. and in that on mouse over, we can display all this information.

Reply:
fixes like this:
![image](https://github.com/user-attachments/assets/cc05df6f-00ec-4cd7-9efd-4821e432a962)

![image](https://github.com/user-attachments/assets/d138a50e-213a-4315-84bb-44496ca98d97)